### PR TITLE
Add "trainremote" command

### PR DESCRIPTION
**Pull request by BillyCheung10botics**
This commit adds a new command "trainremote" that allows users to train their ai model on our AWS server. Currently, the training is hardcoded to train a KerasLinear model with max epoch of 30. You can use this command as the following example shown, 
```
donkey trainremote --tub ~/mycar/data/tub_4_21-06-11/ ~/mycar/data/tub_3_21-06-11/ --path ~/mycar --model modelname 
--url https://hq.robocarstore.com/train/submit_job --get https://hq.robocarstore.com/train/refresh_job_statuses
```
with mandatory arguments `--tub`, `--path`  and `--model`,, and optional arguments `--url` and `--get`
- --tub: specify the path(s) of your data tub(s)
- --path: specify the path of your mycar
- --model: specify the file name (without extension) of the resulting model, loss function plot and movie

- --url: the url to post a training request
- --get: the url to get the status of the submitted job

After submitting a job via the "trainremote", the training takes place on our AWS server with a max. duration of 15 min. You can view your submitted job (and others' jobs) on https://hq.robocarstore.com/train/. Usually, the training takes around 10 min to complete, a model file, a plot and a movie are then downloaded from our server, and saved to the folder 'models' and 'movies' respectively.

### CODE REVIEW

**Comment by DocGarbanzo** [comment]
@BillyCheung10botics - this is a great addition. I have three comments:
1) Can you make the `—path` optional as we have in the `train` and other commands?
2) Can you please convert print statements to logging?
3) The checks are failing, maybe the package is not available in our conda channels, in that case they should go under pip. Can you check that please?

File: donkeycar/management/base.py [hunk 1]
```diff
@@ -464,6 +464,169 @@ def run(self, args):
...
+        filename = self.generate_tub_archive(tub_paths, carapp_path)
+        mp_encoder = MultipartEncoder(
+            fields={
+                'device_id': self.get_wlan_mac_address(self.WLAN),
+                'hostname' : gethostname(),
+                'tub_archive_file': ('file.tar.gz', open(filename, 'rb'), 'application/gzip'),
+                'donkeycar_version': str(__version__)
+            }
+        )
+        print(f"URL submitted to: {submit_job_url}")
```
Comment (DocGarbanzo) [inline donkeycar/management/base.py:539]: Please switch to logging statements instead of print, that's what we are using now.

File: donkeycar/management/base.py [hunk 2]
```diff
@@ -464,6 +464,169 @@ def run(self, args):
...
+        print(f"Data to submit: {mp_encoder}")
+        r = requests.post(
+            submit_job_url,
+            data=mp_encoder,  # The MultipartEncoder is posted as data, don't use files=...!
+            # The MultipartEncoder provides the content-type header with the boundary:
+            headers={'Content-Type': mp_encoder.content_type}
+        )
+        print(f"Submission response: HTTP {r.status_code}")
```
Comment (DocGarbanzo) [inline donkeycar/management/base.py:547]: here too

File: donkeycar/management/base.py [hunk 3]
```diff
@@ -464,6 +464,169 @@ def run(self, args):
...
+        if (r.status_code == 200):
+            # if HTTP 200 OK
+            if ("job_uuid" in r.json()):
+                try:
+                    uuid = r.json()['job_uuid']
+                    print(f"Submitted a Training Job to the remote server {submit_job_url}\n uuid: {uuid}")
```
Comment (DocGarbanzo) [inline donkeycar/management/base.py:553]: and here

File: donkeycar/management/base.py [hunk 4]
```diff
@@ -464,6 +464,169 @@ def run(self, args):
...
+                    return uuid
+                except Exception as e:
+                    print(e)
```
Comment (DocGarbanzo) [inline donkeycar/management/base.py:556]: and here

File: donkeycar/management/base.py [hunk 5]
```diff
@@ -464,6 +464,169 @@ def run(self, args):
...
+                    raise Exception("Failed to call submit job")
+            else:
+                raise Exception("Failed to call submit job")
+        else:
+            raise Exception("Failed to call submit job")
+
+    def get_latest_job_status_from_hq(self, refresh_job_statuses, job_uuids):
+        import requests
+
+        print(f"Getting lastest job status for uuid {job_uuids}")
```
Comment (DocGarbanzo) [inline donkeycar/management/base.py:566]: ...

File: donkeycar/management/base.py [hunk 6]
```diff
@@ -464,6 +464,169 @@ def run(self, args):
...
+            else:
+                model_name = f"job_{uuid}"
+
+            start_time = time.time()
+            run_time = 0
+            checkagain = True
+            while checkagain and (run_time < self.TIMEOUT):
+                result = self.get_latest_job_status_from_hq(url_status, uuid)[0]
+                checkagain = result['status'] == "SCHEDULED" or result['status'] == "TRAINING"
+                print(f"Training Status:{result['status']} at {datetime.now().strftime('%Y-%m-%d, %H:%M:%S')}")
```
Comment (DocGarbanzo) [inline donkeycar/management/base.py:604]: ...

File: donkeycar/management/base.py [hunk 7]
```diff
@@ -464,6 +464,169 @@ def run(self, args):
...
+                    proc = subprocess.Popen(command)
+                    command = ["curl", "--fail", result['model_movie_url'], "--output", f"{car_path}/movies/{model_name}.mp4"]
+                    proc = subprocess.Popen(command)
+                    print("Downloaded model, accuracy graph and movie from the URLs")
+                    break
+                elif result['status'] in ["NO_CAPACITY", "NO_QUOTA", "SPOT_REQUEST_FAILED", "TIMEOUT"]:
+                    raise Exception(f"Failed to train the submitted job\nError : {result['status']}")
+                
+                run_time = time.time()-start_time
+                print(f"Time spent: {run_time} s")
```
Comment (DocGarbanzo) [inline donkeycar/management/base.py:629]: ...

File: install/envs/mac.yml [hunk 8]
```diff
@@ -33,6 +33,9 @@ dependencies:
   - psutil
   - kivy=2.0.0
   - plotly
+  - requests
+  - requests_toolbelt
```
Comment (DocGarbanzo) [inline install/envs/mac.yml:37]: It seems we can't find this package in conda main and conda forge channels, maybe it's only available in pip?

File: install/envs/ubuntu.yml [hunk 9]
```diff
@@ -35,6 +35,9 @@ dependencies:
   - kivy=2.0.0
   - plotly
   - tensorflow==2.2.0
+  - requests
+  - requests_toolbelt
```
Comment (DocGarbanzo) [inline install/envs/ubuntu.yml:39]: This seems to produce the same problem as in the Mac conda yaml file.

File: install/envs/windows.yml [hunk 10]
```diff
@@ -34,6 +34,9 @@ dependencies:
   - kivy=2.0.0
   - plotly
   - psutil
+  - requests
+  - requests_toolbelt
```
Comment (DocGarbanzo) [inline install/envs/windows.yml:38]: No idea if this works in windows... do you have any means to check that?

**Comment by DocGarbanzo** [comment]
@BillyCheung10botics - can you have a look at addressing the installation issues from the checks above, and a couple of minor comments? It would be great to have that change in Donkey Car.

**Review by DocGarbanzo — CHANGES_REQUESTED** [review]
